todo_react/todo-react/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import React, {useState} from 'react';
+import './App.css';
+import {AddTask} from "./components/AddTask";
+import {ToDoItem} from "./components/ToDoItem";
+import {ITodo} from "./models";
+
+export const App: React.FC = () => {
+    let todoList = [...JSON.parse(localStorage.getItem('TodoList') || "")]
+    const [todos, setTodos] = useState<Array<{ id: number, completed: boolean, title: string }>>(todoList)
+    let emptyList = todoList.length === 0 ? "No Tasks added" : ""
+    const addTodo = (todo: ITodo) => {
+        setTodos([...todos, todo])
+        localStorage.setItem('TodoList', JSON.stringify([...todos, todo]))
```

Comment:
I would move logic `localStorage.setItem('TodoList', JSON.stringify(todos))` to separate function:

```typescript
const saveToLocalStorage = (todos: ITodo[]) => {
   localStorage.setItem('TodoList', JSON.stringify(todos))
}
```

Reply:
```typescript
const newTodos = [...todos, todo]
setTodos(newTodos)
saveToLocalStorage(newTodos)
```